### PR TITLE
Settlement reports delete manual adjustment

### PR DESCRIPTION
**Pull request by msbodetti**
### References Issues:
https://app.asana.com/0/1158142652422330/1156925455365674/f
### Description:
Feature for users to remove manual adjustments.
### Release Screenshots / Video:
https://share.getcloudapp.com/bLuGWXep
### Environment Variables:
 * No change

### API requirements:
bn-api-node updates

### CODE REVIEW

File: src/components/pages/admin/reports/settlement/DeleteDialog.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+class DeleteDialog extends Component {
+	constructor(props) {
+		super(props);
+
+		this.defaultState = {
+			isDeleting: false
+		};
+
+		this.state = this.defaultState;
```
Comment (Krakaw) [inline src/components/pages/admin/reports/settlement/DeleteDialog.js]: Why not just set `this.state` directly to the default state?

File: src/components/pages/admin/reports/settlement/DeleteDialog.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+		const { id } = this.props;
+
+		Bigneon()
+			.settlementAdjustments.delete({ id: id })
+			.then(response => {
+				const { status } = response;
+				status === 200
+					? this.adjustmentDeleted()
+					: notifications.showFromErrorResponse({
+						defaultMessage: "Deleting settlement adjustment failed."
```
Comment (Krakaw) [inline src/components/pages/admin/reports/settlement/DeleteDialog.js]: Let's start making friendlier error messages maybe `There was an error deleting the settlement adjustment, please try again`

File: src/components/pages/admin/reports/settlement/DeleteDialog.js (hunk 1 above)
Comment (msbodetti) [reply src/components/pages/admin/reports/settlement/DeleteDialog.js]: Thanks. It must have been there where there were more than one state.

File: src/components/pages/admin/reports/settlement/DeleteDialog.js (hunk 2 above)
Comment (msbodetti) [reply src/components/pages/admin/reports/settlement/DeleteDialog.js]: Will take a note of this. Thanks!

**Comment by msbodetti** [comment]
Hi @Krakaw I've tested on my local with the new dependancies and it works as expected. I can't seem to build new scratches for testing on there.

**Comment by msbodetti** [comment]
Hi @Krakaw I created a scratch but realised that the settlement reports have not been generated since it's a new deploy. I tested it on local with dev api and it's working as expected.

File: src/components/pages/admin/reports/settlement/DeleteDialog.js [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+		};
+
+		this.onDeleteAdjustment = this.onDeleteAdjustment.bind(this);
+	}
+
+	static getDerivedStateFromProps(props, state) {
+		const { id } = props;
+
+		return { id };
+	}
```
Comment (Krakaw) [inline src/components/pages/admin/reports/settlement/DeleteDialog.js]: Unused

File: src/components/pages/admin/reports/settlement/DeleteDialog.js [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+	onDeleteAdjustment() {
+		const { id } = this.props;
+
+		Bigneon()
+			.settlementAdjustments.delete({ id: id })
```
Comment (Krakaw) [inline src/components/pages/admin/reports/settlement/DeleteDialog.js]:
```suggestion
			.settlementAdjustments.delete({ id })
```

**Pull request merged**